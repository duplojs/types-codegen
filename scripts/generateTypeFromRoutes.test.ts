import { makeResponseContract, OkHttpResponse, useBuilder, zod } from "@duplojs/core";
import { generateTypeFromRoutes } from "./generateTypeFromRoutes";

it("generateTypeFromRoutes", () => {
	const route = useBuilder()
		.createRoute("GET", "/test")
		.extract({
			query: {
				page: zod.number(),
			},
		})
		.handler(
			(pickup) => new OkHttpResponse("test", pickup("page")),
			makeResponseContract(OkHttpResponse, "test", zod.number()),
		);

	expect(
		generateTypeFromRoutes([route]),
	)
		.toMatchSnapshot();
});
