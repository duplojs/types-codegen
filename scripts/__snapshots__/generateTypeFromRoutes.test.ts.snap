// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateTypeFromRoutes 1`] = `
"type Routes = {
    method: "GET";
    path: "/test";
    query: {
        page: number;
    };
    response: {
        code: 200;
        information: "test";
        body: number;
        ok: true;
    };
};

export { Routes };"
`;
